### PR TITLE
Bluetooth: controller: hci: Add user hooks.

### PR DESCRIPTION
**Pull request by wopu-ot**
Add hooks for processing of HCI user events on the controller side.

### CODE REVIEW

**Review by cvinayak — CHANGES_REQUESTED** [review]
~~LGTM, except could you add CONFIG_BT_CTLR_USER_EXT to https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/bluetooth/init/prj_controller_dbg_ll_sw_split.conf  or similar, so that CI can build it?~~

**Review by cvinayak — APPROVED** [review]
LGTM

**Review by thoh-ot — APPROVED** [review]

File: subsys/bluetooth/controller/hci/hci_user_ext.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ *  @brief HCI controller user extensions
+ */
+
+/*
+ * Copyright (c) 2019 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+s8_t hci_get_class_proprietary(struct node_rx_pdu *node_rx);
```
Comment (carlescufi) [inline subsys/bluetooth/controller/hci/hci_user_ext.h]: Can we use `_user`  everywhere? Also maybe better:

`hci_get_user_class()`

File: subsys/bluetooth/controller/hci/hci_user_ext.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+void hci_encode_control_proprietary(struct node_rx_pdu *node_rx, struct pdu_data *pdu_data, struct net_buf *buf);
```
Comment (carlescufi) [inline subsys/bluetooth/controller/hci/hci_user_ext.h]: `hci_encode_user_control()`

**Review by carlescufi — CHANGES_REQUESTED** [review]
Thanks! Some naming issues

**Comment by cvinayak** [comment]
I second @carlescufi on the namings

**Comment by wopu-ot** [comment]
Naming updated to `hci_user_ext_get_class()` and `hci_user_ext_encode_control()` as discussed on slack.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**